src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -321,7 +326,8 @@ public Txn createTxn(ClientState state, QueryOptions options)
             List<TxnNamedRead> reads = createNamedReads(options, state, ImmutableMap.of(), keySet::add);
             Keys txnKeys = toKeys(keySet);
             TxnRead read = createTxnRead(reads, txnKeys, null);
-            return new Txn.InMemory(txnKeys, read, TxnQuery.ALL);
+            Txn.Kind kind = txnKeys.size() == 1 && getNonSerialWriteStrategy() == accord ? EphemeralRead : Read;
```

Comment:
Minor, but this is a per key property that wouldn't require absolutely all writes going through Accord. Would be nice to have the performance benefit during migration, but maybe a TODO.